### PR TITLE
Fix docs for TransformerBlock and remove incompatible functions

### PR DESCRIPTION
**Pull request by mashu**
- Added examples how to use TransformerBlock (not just how to use constructor)
- Removed confusing cross_att_padding_mask and self_att_padding_mask which were returning in log-space and couldn't be used with apply_pad_mask. Now mask can be passed directly.

### CODE REVIEW

**Comment by murrellb** [comment]
Note: self_att_padding_mask and cross_att_padding_mask are used with the IPA layers, which (currently) need the full pairwise mask to be materialized, as it gets added into the attention logits prior to softmax:
image
We can't remove these, but we could maybe improve the docstrings?
